registration/include/pcl/registration/impl/transformation_estimation_symmetric_point_to_plane_lls.hpp

```diff
@@ -222,30 +196,12 @@ estimateRigidTransformation (ConstCloudIterator<PointSource>& source_it, ConstCl
     ATb.coeffRef (3) += nx * d;
     ATb.coeffRef (4) += ny * d;
     ATb.coeffRef (5) += nz * d;
-
-    ++target_it;
-    ++source_it;    
   }
-  ATA.coeffRef (6) = ATA.coeff (1);
-  ATA.coeffRef (12) = ATA.coeff (2);
-  ATA.coeffRef (13) = ATA.coeff (8);
-  ATA.coeffRef (18) = ATA.coeff (3);
-  ATA.coeffRef (19) = ATA.coeff (9);
-  ATA.coeffRef (20) = ATA.coeff (15);
-  ATA.coeffRef (24) = ATA.coeff (4);
-  ATA.coeffRef (25) = ATA.coeff (10);
-  ATA.coeffRef (26) = ATA.coeff (16);
-  ATA.coeffRef (27) = ATA.coeff (22);
-  ATA.coeffRef (30) = ATA.coeff (5);
-  ATA.coeffRef (31) = ATA.coeff (11);
-  ATA.coeffRef (32) = ATA.coeff (17);
-  ATA.coeffRef (33) = ATA.coeff (23);
-  ATA.coeffRef (34) = ATA.coeff (29);
 
   // Solve A*x = b
-  Vector6d x = static_cast<Vector6d> (ATA.inverse () * ATb);
+  Vector6d x = static_cast<Vector6d> (ATA.selfadjointView<Eigen::Upper> ().ldlt ().solve (ATb));
```

Comment:
ATA is a least-squares matrix and therefore symmetric positive semidefinite. If it's strictly positive definite, the Cholesky decomposition will be more efficient than inverting the matrix because it allows you to take advantage of the symmetric structure. If its semidefinite and has zero eigenvalues, both the decomposition and the eigenvalues will theoretically fail. However, this is probably such a rare case that the additional cost of a rank-revealing decomposition is not worth it.